src/networks/build_attention.py

```diff
@@ -3,86 +3,69 @@
 import numpy as np
 
 
-_CHR_IDX = string.ascii_lowercase
+def build_quadratic_attention_equation(rank, attn_axes):
+    batch_dims = _get_index_of_batch_dims(rank, attn_axes)
+    query_notation, value_notation = _get_query_and_value_notation(rank, batch_dims)
```

Comment:
this code is reusable for quadratic and linear methods, so put them into one function